File: NationalPokedex/NationalPokedex/Utils/Pokemon.swift
//
//  PokeDetail.swift
//  NationalPokedex
//
//  Created by Akihiro Matsuyama on 2024/04/14.
//

import Foundation

/// 個体情報と種情報をまとめたポケモンデータ
/// HashableはNavigationStackでパスをハッシュ化するのに使用
struct Pokemon: Identifiable, Codable, Hashable{
    let id: Int
    var individual: PokeIndividual
    var species: PokeSpecies
}

extension Pokemon {
    // フシギダネ
    static let mock = Pokemon(id: 1,
                              individual: PokeIndividual.mock,
                              species: PokeSpecies.mock)
}

File: NationalPokedex/shared/src/commonMain/kotlin/com/akidon0000/nationalpokedex/Model/Pokemon.kt
package com.akidon0000.nationalpokedex.Model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Pokemon(
    @SerialName("id")
    val id: Int,

    @SerialName("individual")
    var individual: PokeIndividual,

    @SerialName("species")
    var species: PokeSpecies
) {
    companion object {
        // フシギダネ
        val mock = Pokemon(
            id = 1,
            individual = PokeIndividual.mock,
            species = PokeSpecies.mock
        )
    }
}
